test/lib/ansible_test/_internal/sanity/import.py

```diff
@@ -92,91 +103,112 @@ def test(self, args, targets, python_version):
-        importer_path = os.path.join(virtual_environment_bin, 'importer.py')
-        yaml_to_json_path = os.path.join(virtual_environment_bin, 'yaml_to_json.py')
-        if not args.explain:
-            os.symlink(os.path.abspath(os.path.join(SANITY_ROOT, 'import', 'importer.py')), importer_path)
-            os.symlink(os.path.abspath(os.path.join(SANITY_ROOT, 'import', 'yaml_to_json.py')), yaml_to_json_path)
-
-        # activate the virtual environment
-        env['PATH'] = '%s:%s' % (virtual_environment_bin, env['PATH'])
-
-        env.update(
-            SANITY_TEMP_PATH=ResultType.TMP.path,
-        )
-
-        if data_context().content.collection:
-            env.update(
-                SANITY_COLLECTION_FULL_NAME=data_context().content.collection.full_name,
-                SANITY_EXTERNAL_PYTHON=python,
-            )
-
-        virtualenv_python = os.path.join(virtual_environment_bin, 'python')
-        virtualenv_pip = generate_pip_command(virtualenv_python)
-
-        # make sure coverage is available in the virtual environment if needed
-        if args.coverage:
-            run_command(args, generate_pip_install(virtualenv_pip, '', packages=['setuptools']), env=env, capture=capture_pip)
-            run_command(args, generate_pip_install(virtualenv_pip, '', packages=['coverage']), env=env, capture=capture_pip)
-
-        try:
-            # In some environments pkg_resources is installed as a separate pip package which needs to be removed.
-            # For example, using Python 3.8 on Ubuntu 18.04 a virtualenv is created with only pip and setuptools.
-            # However, a venv is created with an additional pkg-resources package which is independent of setuptools.
-            # Making sure pkg-resources is removed preserves the import test consistency between venv and virtualenv.
-            # Additionally, in the above example, the pyparsing package vendored with pkg-resources is out-of-date and generates deprecation warnings.
-            # Thus it is important to remove pkg-resources to prevent system installed packages from generating deprecation warnings.
-            run_command(args, virtualenv_pip + ['uninstall', '--disable-pip-version-check', '-y', 'pkg-resources'], env=env, capture=capture_pip)
-        except SubprocessError:
-            pass
-
-        run_command(args, virtualenv_pip + ['uninstall', '--disable-pip-version-check', '-y', 'setuptools'], env=env, capture=capture_pip)
-        run_command(args, virtualenv_pip + ['uninstall', '--disable-pip-version-check', '-y', 'pip'], env=env, capture=capture_pip)
-
-        cmd = ['importer.py']
+        results = []
 
-        data = '\n'.join(paths)
+        for import_type, test, add_ansible_requirements in (
+                ('module', _get_module_test(True), False),
+                ('plugin', _get_module_test(False), True),
+        ):
+            if import_type == 'plugin' and python_version.startswith('2.6'):
```

Comment:
```suggestion
            if import_type == 'plugin' and python_version in REMOTE_ONLY_PYTHON_VERSIONS:
```

Also, as a side-note, we're going to need something more complex to handle collections which have controller-only modules and module_utils -- but that's not something we should try to address in this PR.

Reply:
Implemented in 0a631981cd731f62acf0eb33fd68eb3e41c7b482.